src/gatsby-node.js

```diff
@@ -1,35 +1,132 @@
 const {
-        normalizeFields, 
         normalizeEntry, 
         processContentType, 
-        processEntry
+        processEntry,
+        processAsset,
+        makeEntryNodeUid,
+        makeAssetNodeUid
     } = require("./normalize");
 
 const fetchData = require("./fetch");
 
-exports.sourceNodes = async ({ boundActionCreators, createNodeId },configOptions) => {
-    const { createNode } = boundActionCreators;
 
-    const { contentstackData } = await fetchData(configOptions);
+exports.sourceNodes = async ({ actions, getNode, getNodes, createNodeId, store, reporter },configOptions) => {
+    const { createNode, deleteNode, touchNode, setPluginStatus } = actions;
+    let syncToken 
+
+    if (
+        store.getState().status.plugins &&
+        store.getState().status.plugins[`gatsby-source-contentstack`] &&
+        store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ]
+    ) {
+        syncToken = store.getState().status.plugins[`gatsby-source-contentstack`][
+            `contentstack-sync-token-${configOptions.api_key}`
+        ];
+    }
 
+    configOptions.syncToken = syncToken || null;
     
-    contentstackData.contentTypes.forEach(contentType => {
-        const entries = contentstackData.entries[contentType.uid];
-
-        entries.forEach(entry => {
-            const normalizedEntry = normalizeEntry(contentType, entry, contentstackData.entries, createNodeId);
-            // Process the contentTypes data to match the structure of a Gatsby node
-            const entryNode = processEntry(contentType, normalizedEntry, createNodeId);
-            // Use Gatsby's createNode helper to create a node from the node data
-            createNode(entryNode);
+    const { contentstackData } = await fetchData(configOptions);
+    
+    let publishedEntriesType   = contentstackData.syncData.filter(item => item.type === 'entry_published') || []; 
```

Comment:
Why not do this grouping once? e.g.

```js
const data = contentstackData.syncData.reduce((merged, item) => {
  if (!merged[item.type]) {
    merged[item.type] = []
  }
  merged[item.type].push(item)
  return merged
}, {})
```

then you can do data['entry_unpublished'] to get unpublished (and other types)

Reply:
Done